### PR TITLE
docs: Remove last references to CLC

### PR DESCRIPTION
**Pull request by pothos**
The transpilation is now done for Butane only.
Also fix a broken link.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ terraform {
   required_providers {
     ct = {
       source  = "poseidon/ct"
-      version = "0.11.0"
```
Comment (dghubble) [inline README.md:17]: Let's wait until closer to release before starting to change the version

File: docs/data-sources/ct_config.md [hunk 2]
```diff
@@ -21,15 +21,14 @@ resource "aws_instance" "worker" {
...
 
 ## Argument Reference
 
-* `content` - contents of a Container Linux Config (CLC) or Fedora CoreOS Config (FCC) that should be validated and transpiled to Ignition.
+* `content` - contents of a Butane Config that should be validated and transpiled to Ignition.
 * `strict` - strictly treat validation warnings as errors (default: false).
 * `pretty_print` - indent transpiled Ignition for visual prettiness (default: false)
-* `snippets` - list of Container Linux Config (CLC) or Fedora CoreOS Config (FCC) snippets to merge into the content. For FCCs, content and snippet configs must have the same `version`.
-* `platform` - (Container Linux only) - enable [platform-specific](https://github.com/coreos/container-linux-config-transpiler/blob/master/config/platform/platform.go) susbstitutions
+* `snippets` - list of Container Butane snippets to merge into the content. Content and snippet configs must have the same `version` and `variant`.
```
Comment (dghubble) [inline docs/data-sources/ct_config.md:31]: Container Butane -> Butane

**Review by dghubble — COMMENTED** [review]
small things

File: README.md (hunk 1 above)
Comment (pothos) [reply README.md]: Ok, remove the change

File: docs/data-sources/ct_config.md (hunk 2 above)
Comment (pothos) [reply docs/data-sources/ct_config.md]: Thanks

**Pull request merged**